pkg/netdevice/pciNetDevice.go

```diff
@@ -90,13 +94,27 @@ func NewPciNetDevice(dev *ghw.PCIDevice, rFactory types.ResourceFactory, rc *typ
 		linkType = la.EncapType
 	}
 
+	devlinkClient := utils.NewDevlinkInfoClient()
+
+	var ddpFunc ddpProfileGetFunc
+
+	if devlinkClient.IsDevlinkSupportedByPCIDevice(pciAddr) {
```

Comment:
If this would be a part of `GetDDPProfiles` it would be invoked each time the profile is queried for the device. Now the support will be only checked when the device object is being created. I think this could be discussed.